geozero/src/csv/csv_error.rs

```diff
@@ -0,0 +1,14 @@
+//! CSV Error type.
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum CsvError {
+    #[error("column not found or null")]
+    ColumnNotFound,
+    #[error("Invalid UTF-8 encoding")]
+    InvalidUtf8,
+    #[error("error processing dataset: `{0}`")]
+    Processing(String),
```

Comment:
This seems unused?

Reply:
Hi Michael,

I wanted to use the `InvalidUtf8` variant in `csv/mod.rs` to replace `GeozeroError::Geometry`, however `CsvError` will not convert to `GeozeroError` to match the custom `Result` type. I created the `Processing` variant to replace `GeozeroError::Dataset` on line 217 of `csv_reader.rs`, but I'm also running into a similar problem there.